### PR TITLE
feat: CRN-1050 Create Prod/Dev Contentful Apps

### PR DESCRIPTION
**Pull request by peterstarling**
Adding the following:
* Production and Development Contentful schema sync on master pipeline
* GraphQL Schema sync commands

### CODE REVIEW

File: .github/workflows/reusable-build.yml [hunk 1]
```diff
@@ -17,6 +17,9 @@ on:
       LIGHTSTEP_TOKEN:
         description: 'Lightstep Token'
         required: true
+      CONTENTFUL_MANAGEMENT_TOKEN:
+        description: 'Contentful Management Token'
+        required: true
```
Comment (markcsims) [inline .github/workflows/reusable-build.yml:22]: This isn't used anywhere

File: .github/workflows/reusable-deployment.yml [hunk 2]
```diff
@@ -188,6 +191,35 @@ jobs:
...
+        uses: ./.github/actions/cache-unplugged
+        with:
+          environment-name: ${{ inputs.environment-name }}
+      - name: Synchronize Contentful
+        run: |
+          yarn workspace @asap-hub/contentful schema:import --management-token=$CONTENTFUL_MANAGEMENT_TOKEN
+        env:
+          CONTENTFUL_MANAGEMENT_TOKEN: ${{ secrets.CONTENTFUL_MANAGEMENT_TOKEN }}
+          CONTENTFUL_SPACE_ID: ${{ steps.setup.outputs.contentful-space-id }}
+          CONTENTFUL_ENVIRONMENT: ${{ inputs.environment-name }}
```
Comment (markcsims) [inline .github/workflows/reusable-deployment.yml:222]: This is tied to our environment name. branch, development and production. 

Might be better to separate this off into it's own variable

File: packages/contentful/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  extends: ['@asap-hub/eslint-config-asap-hub'],
+  rules: {
+    'lines-between-class-members': 'off',
+    'max-classes-per-file': 'off',
```
Comment (markcsims) [inline packages/contentful/.eslintrc.js:5]: Why are we turning these 2 things off?

File: packages/contentful/package.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    "node": "16.x"
+  },
+  "module": "build/index.js",
+  "main": "build-cjs/index.js",
+  "types": "build/index.d.ts",
+  "scripts": {
+    "build:babel": "../../scripts/build-babel.sh build",
+    "watch:babel": "../../scripts/build-babel.sh watch",
+    "schema:export": "contentful space export --skip-content=true --space-id=$CONTENTFUL_SPACE_ID --environment-id=$CONTENTFUL_ENVIRONMENT --content-file=./schema/schema.json",
+    "schema:import": "contentful space import --content-model-only=true --space-id=$CONTENTFUL_SPACE_ID --environment-id=$CONTENTFUL_ENVIRONMENT --content-file=./schema/schema.json"
```
Comment (markcsims) [inline packages/contentful/package.json:15]: Should we start prefixing/suffexing these things with crn/gp2? Even the schema location in the package.

**Review by markcsims — CHANGES_REQUESTED** [review]

File: .env.example [hunk 5]
```diff
@@ -19,3 +19,5 @@ REACT_APP_GP2_AUTH0_AUDIENCE=https://dev.gp2.asap.science
 REACT_APP_GP2_AUTH0_CLIENT_ID=sUSMSrBS3FMhb58bxiwQZOTPbZMPWIje
 REACT_APP_GP2_AUTH0_DOMAIN=dev-gp2-hub.us.auth0.com
 SQUIDEX_BASE_URL=https://cloud.squidex.io
+CONTENTFUL_SPACE_ID=5v6w5j61tndm
+CONTENTFUL_ENVIRONMENT=dev
```
[the remark this answers is not in the document]
Comment (peterstarling) [reply .env.example:23]: it's for local development - the command in package.json uses it

File: packages/contentful/package.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  "engines": {
+    "node": "16.x"
+  },
+  "module": "build/index.js",
+  "main": "build-cjs/index.js",
+  "types": "build/index.d.ts",
+  "scripts": {
+    "build:babel": "../../scripts/build-babel.sh build",
+    "watch:babel": "../../scripts/build-babel.sh watch",
+    "schema:export:crn": "contentful space export --skip-content=true --space-id=$CONTENTFUL_SPACE_ID --environment-id=$CONTENTFUL_ENVIRONMENT --content-file=./schema/crn-schema.json",
```
Comment (peterstarling) [inline packages/contentful/package.json:14]:
```suggestion
    "space:export:crn": "contentful space export --skip-content=true --space-id=$CONTENTFUL_SPACE_ID --environment-id=$CONTENTFUL_ENVIRONMENT --content-file=./schema/crn-schema.json",
```

File: packages/contentful/package.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    "schema:import:crn": "contentful space import --content-model-only=true --space-id=$CONTENTFUL_SPACE_ID --environment-id=$CONTENTFUL_ENVIRONMENT --content-file=./schema/crn-schema.json",
```
Comment (peterstarling) [inline packages/contentful/package.json:15]:
```suggestion
    "space:import:crn": "contentful space import --content-model-only=true --space-id=$CONTENTFUL_SPACE_ID --environment-id=$CONTENTFUL_ENVIRONMENT --content-file=./schema/crn-schema.json",
```

File: packages/contentful/src/autogenerated-gql/gql.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable */
```
Comment (peterstarling) [inline packages/contentful/src/autogenerated-gql/gql.ts:1]: move this file to crn/

**Comment by peterstarling** [comment]
add documentation

**Review by markcsims — APPROVED** [review]

**Pull request merged**